src/cocotb/_scheduler.py

```diff
@@ -172,6 +173,12 @@ def thread_wait(self):
         return self.state
 
 
+class GPIState(Enum):
+    NORMAL = auto()
+    READ_WRITE = auto()
+    READ_ONLY = auto()
+
+
 class Scheduler:
```

Comment:
Maybe change the uses of "mode" to "state" in this docstring?

Reply:
I rewrote the docstring since it could use a bit more detail.